.github/workflows/e2e.yml

```
@@ -28,7 +28,80 @@ concurrency:
+      - name: Inject go cache into docker
+        uses: reproducible-containers/buildkit-cache-dance@v3.1.2
+        with:
+          cache-map: |
+            {
+              "go-cache": "/root/.cache/go-build"
+            }
+          skip-extraction: ${{ steps.restore-go-cache.outputs.cache-hit || github.event_name != 'push' }}
+
+      # this ensures that the version is consistent between cache build and make build
+      - name: Set version for cache
+        run: |
```

actionlint

shellcheck reported issue in this script: SC2086:info:2:38: Double quote to prevent globbing and word splitting

(shellcheck)